app/controllers/internal_api/v1/pillars_controller.rb

```diff
@@ -20,13 +20,22 @@ def pillar_contents
   end
 
   def registry_contents
-    registries = DockerRegistry.all.map do |reg|
-      {
-        url:         reg.url,
-        mirror:      reg.mirror,
-        certificate: (reg.certificate.present? ? reg.certificate.certificate : nil)
-      }
+    registries = DockerRegistry.is_registry.map do |reg|
+      url:  reg.url,
+      cert: (reg.certificate.present? ? reg.certificate.certificate : nil)
     end
-    { registries: registries }
+    registry_mirrors = []
+    DockerRegistry.is_mirror.group(:mirror).pluck(:mirror).each do |mirror|
```

Comment:
Rename `|mirror|` to something like  `|remote_registry|` just to make it clear that is the url of a registry and not a mirror.